flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/StreamPlanner.scala

```diff
@@ -155,7 +153,13 @@ class StreamPlanner(
       case catalogSink: CatalogSinkModifyOperation =>
         getTableSink(catalogSink.getTablePath)
           .map(sink => {
-            TableSinkUtils.validateSink(catalogSink.getChild, catalogSink.getTablePath, sink)
+            TableSinkUtils.validateSink(catalogSink.getStaticPartitions,
+              catalogSink.getChild, catalogSink.getTablePath, sink)
+            sink match {
+              case partitionableSink: PartitionableTableSink =>
```

Comment:
Only set static partitions when sink has partitioned fields, i.e. is a partitioned sink. Currently, we set a partition map always which may have unexpected effects.